spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ContextToSourceData.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.fabric8.config;
+
+import java.util.function.Function;
+
+import org.springframework.cloud.kubernetes.commons.config.SourceData;
+
+/**
+ * A more succinct way to define a Function from Fabric8ConfigContext to SourceData.
+ *
+ * @author wind57
+ */
+interface Fabric8ContextToSourceData extends Function<Fabric8ConfigContext, SourceData> {
```

Comment:
Any reason to not make these public?

Reply:
not really :) I was just going nuts to hide everything :D not sure if I am correct in this case or not. I'll listen to your advise